### PR TITLE
Adapt perl script to handle multiple .desc files

### PR DESCRIPTION
**Pull request by LAJW**
Solves the following issue: https://github.com/diffblue/test-gen/issues/680

Changes test.pl to name output *.out files after *.desc instead
of *.class files. Also updates failed-test-printer.pl to work with
those new *.out files.

### CODE REVIEW

**Comment by tautschnig** [comment]
Is this the same as #1082, except that it's now for master?

**Comment by tautschnig** [comment]
FWIW, I'm rather confused: only with #1082 in place there is a renaming according to test.desc files!?

**Comment by allredj** [comment]
This replaces #1082 (which has been closed). It does the same renaming plus corrects the problem with failed tests cases.
We first apply the fix on master and then paste the fix on test-gen-support.

**Comment by tautschnig** [comment]
> It does the same renaming plus corrects the problem with failed tests cases.

Can the commit message please be amended to state this, rather than claiming the converse renaming?

**Comment by allredj** [comment]
@tautschnig I don't understand your comment. To me the commit message expresses exactly what it does. Naming the files after the `.desc` files prevents overwriting the `.out` in case we have several of theses `.desc`.

**Comment by tautschnig** [comment]
Ok, maybe that's just a problem of the choice of tense: "test.pl names output *.out files after *.desc instead of *.java files" to me suggests that this is the current state, before the commit. It's not my native language, so I may be very wrong. How about "Modify test.pl to name output ..." (and then please don't refer to .java files, because these are never used as input).

File: regression/failed-tests-printer.pl [hunk 1]
```diff
@@ -6,20 +6,22 @@
...
 while (<LOG>) {
   chomp;
   if (/^Test '(.+)'/) {
     $current_test = $1;
     $printed_this_test = 0;
+  } elsif (/Output:\s+([^\s]+)/) {
+    $output_file = $1;
   } elsif (/\[FAILED\]\s*$/) {
     if(0 == $printed_this_test) {
+      print "'$current_test/$output_file'";
```
Comment (tautschnig) [inline regression/failed-tests-printer.pl]: Is this debug output? Note that the two blank lines printed next are meant to separate tests, so it may be better to place this output line further below (if it is meant to stay).

**Comment by LAJW** [comment]
Commit has been amended. Debug print removed and description updated. It's hopefully clearer now.

**Comment by allredj** [comment]
@tautschnig My bad. I couldn't see your approval for some reason so I re-requested it.

File: regression/failed-tests-printer.pl [hunk 2]
```diff
@@ -6,20 +6,21 @@
...
   } elsif (/\[FAILED\]\s*$/) {
     if(0 == $printed_this_test) {
       $printed_this_test = 1;
       print "\n\n";
       print "Failed test: $current_test\n";
-      my $outf = `sed -n '2p' $current_test/test.desc`;
-      $outf =~ s/\..*$/.out/;
-      system("cat $current_test/$outf");
+      system("cat $current_test/$output_file");
       print "\n\nFailed test.desc lines:\n";
```
Comment (peterschrammel) [inline regression/failed-tests-printer.pl]: `test.desc` should be replaced by the actual name of the test descriptor

Comment (LAJW) [inline regression/failed-tests-printer.pl]: I'm not sure what you do mean by the "test descriptor". "foo.desc" will spawn "foo.out" in the same directory, "bar.desc" -> "bar.out" etc. Should I go through all repositories and rename test-name/test.desc to test-name/test-name.desc? That seems like a structural change and would require a separate ticket.

Comment (tautschnig) [inline regression/failed-tests-printer.pl]: It's about the literal use of `test.desc` in that `print` statement on line 24. Those files, after all, need not actually be named "test.desc", but could be "test1.desc" etc (your commit says "handle multiple .desc files"). Hence that `print` statement should say, e.g., "Failed test1.desc lines:".

Comment (LAJW) [inline regression/failed-tests-printer.pl]: Thanks for noticing that. Fixed.

File: regression/failed-tests-printer.pl [hunk 3]
```diff
@@ -6,21 +6,22 @@
...
     if(0 == $printed_this_test) {
       $printed_this_test = 1;
       print "\n\n";
       print "Failed test: $current_test\n";
-      my $outf = `sed -n '2p' $current_test/test.desc`;
-      $outf =~ s/\..*$/.out/;
-      system("cat $current_test/$outf");
-      print "\n\nFailed test.desc lines:\n";
+      system("cat $current_test/$output_file");
+      print "\n\nFailed $output_file lines:\n";
```
Comment (tautschnig) [inline regression/failed-tests-printer.pl]: Is that really what the output holds? I thought the output listed lines from *.desc that could _not_  be found in `$output_file`?!

File: regression/strings/StringCompare01/test.desc [hunk 4]
```diff
@@ -6,3 +6,5 @@ StringCompare01.class
 ^VERIFICATION SUCCESSFUL$
 --
 ^warning: ignoring
+--
+Fails because string.regionMatches is not implemented
```
Comment (tautschnig) [inline regression/strings/StringCompare01/test.desc]: How is this related to the commit?

Comment (LAJW) [inline regression/strings/StringCompare01/test.desc]: My apologies, multitasking issues.

**Comment by LAJW** [comment]
OK, Now it should be fine. Added additional "Descriptor" field to the tests.log file.
